content/artifacts/harvest-artifacts/plugin.js

```diff
@@ -0,0 +1,351 @@
+    stepper.value = `${this.numberCount}`;
+    stepper.style.width = '80%';
+    stepper.style.height = '24px';
+
+    let stepperLabel = document.createElement('label');
+    stepperLabel.innerText = `How many to harvest:`;
+    stepperLabel.style.display = 'block';
+
+    let counter = document.createElement('span');
+    counter.innerText = `${stepper.value}`;
+    counter.style.float = 'right';
+
+    stepper.onchange = (evt) => {
+      counter.innerText = `${evt.target.value}`;
+      try {
+        this.numberCount = parseInt(evt.target.value, 10);
+      } catch (e) {
+        console.error('could not parse number counter', e);
+      }
+    }
+
+    let levelLabel = document.createElement('label');
+    levelLabel.innerText = 'Minimum level:';
+    levelLabel.style.display = 'block';
+
+    let level = document.createElement('select');
+    level.style.background = 'rgb(8,8,8)';
+    level.style.width = '100%';
+    level.style.marginTop = '10px';
+    level.style.marginBottom = '10px';
+
+    popolateLevel();
+    level.value = `${this.minPlanetLevel}`;
+
+    level.onchange = (evt) => {
+      try {
+        this.minPlanetLevel = parseInt(evt.target.value, 10);
+      } catch (e) {
+        console.error('could not parse planet level', e);
+      }
+    }
+
+    let message = document.createElement('div');
+    let failedPlanets = document.createElement('div');
+
+    let autoDeactivateCheckbox = document.createElement('input');
+    autoDeactivateCheckbox.type = "Checkbox";
+    autoDeactivateCheckbox.style.marginLeft = "15px";
+    autoDeactivateCheckbox.checked = autoDeactivate;
+    autoDeactivateCheckbox.id = "autoDeactivateCheckbox";
+
+    let checkBoxLabel = document.createElement('label');
+    checkBoxLabel.setAttribute('for','autoDeactivateCheckbox');
+    checkBoxLabel.innerText = "Auto Deactivate:";
+
+    function setAutoDeactivateCheckbox() {
+      if(autoDeactivateCheckbox.checked == true) 
+        autoDeactivate = true;
+      else {
+        autoDeactivate = false;
+      }
+    }    
+
+    let button = document.createElement('button');
+    button.style.width = '100%';
+    button.style.marginBottom = '10px';
+    button.innerHTML = 'Harvest Artifacts'
+    button.onclick = () => {
+      message.innerText = `Harvesting ${this.numberCount} planets.`;
+      setAutoDeactivateCheckbox();
+
+      let moves = autoProcessArtifacts(
+        false, // true = "showOnly" for debug
+        this.numberCount,
+        this.minPlanetLevel,
+        3,  //this is not used
+        autoDeactivate
+      )
+        .then(errorPs => {
+          failedPlanets.innerText = `${errorPs.length} needs intervention:`;
+          for (let p of errorPs) {
+            const pEntry = document.createElement('div');
+            failedPlanets.appendChild(pEntry);
+            pEntry.innerText = p;
+            pEntry.onclick = () => { ui.centerLocationId(p) };
+          }
+        })
+    }
+
+    setInterval(() => {
```

Comment:
This interval needs to be held so it can be cleared during the destroy phase. A bunch of plugins do this so you should be able to find an example :)